### PR TITLE
Add tests to continuum-auth

### LINKED ISSUE #309: Add tests to continuum-auth
There are no tests which lead to #300

### PR DESCRIPTION
**Pull request by nlisgo**
Closes #309 

This will be a followup to #438

### CODE REVIEW

File: continuum-auth/src/use-cases/login.test.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+describe('login', () => {
+    it.only('redirects', () => {
+        const request = ({
+            params: { token: 'token' },
+        } as unknown) as Request;
+        const response = ({
+            redirect: jest.fn(),
+        } as unknown) as Response;
+
+        config.auth.login_redirect_url = 'http://login_redirect_url';
```
Comment (nlisgo) [inline continuum-auth/src/use-cases/login.test.ts]: This doesn't seem to update the value. Getting the following error when running the test:

```
➜  continuum-auth git:(309-continuum-auth-tests) ✗ yarn run test ./src/use-cases/login.test.ts
yarn run v1.19.1
$ jest ./src/use-cases/login.test.ts
 FAIL  src/use-cases/login.test.ts
  login
    ✕ redirects (6ms)

  ● login › redirects

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    Expected: "http://login_redirect_url"
    Received: "undefined:undefined/submit"

    Number of calls: 1

      16 |         Login(request, response);
      17 |         expect(response.redirect).toHaveBeenCalledTimes(1);
    > 18 |         expect(response.redirect).toHaveBeenCalledWith('http://login_redirect_url');
         |                                   ^
      19 |     });
      20 | });
      21 | 

      at Object.<anonymous> (src/use-cases/login.test.ts:18:35)

{"level":30,"time":1572425268783,"pid":53884,"hostname":"Nathans-MBP-2","service":"continuum-auth","realm":"domai
n","msg":"loginRedirect {\"login_redirect_url\":\"undefined:undefined/submit\"}","v":1}
Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        2.313s
Ran all test suites matching /.\/src\/use-cases\/login.test.ts/i.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

File: continuum-auth/src/repo/profiles.test.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+jest.mock('node-fetch');
+jest.mock('../logger');
+
+describe('profiles', () => {
+    it('fetchs a profile by id', () => {
+        const profilesService = new ProfilesService('http://profiles_service_url');
+        profilesService.getProfileById('profile-id');
+        expect(fetch).toHaveBeenCalledTimes(1);
+        expect(fetch).toHaveBeenCalledWith('http://profiles_service_url/profile-id');
```
Comment (nlisgo) [inline continuum-auth/src/repo/profiles.test.ts]: I would like to test the handling of a response from `fetch` within the `getProfileById` method.

File: continuum-auth/src/jwt.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    default: {
+        journal_jwt: {
+            secret: 'secret',
+        },
+    },
+}));
+
+describe('jwt', () => {
+    describe('decodeJournalToken', () => {
+        beforeEach(() => jest.resetAllMocks());
```
Comment (erezmus) [inline continuum-auth/src/jwt.test.ts]: You can do:
```
beforeEach(jest.resetAllMocks);
```

File: continuum-auth/src/use-cases/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Request, Response } from 'express';
+import { HealthCheck } from './index';
+
+jest.mock('../logger');
+
+describe('healthcheck', () => {
+    it.only('is OK', () => {
```
Comment (will-byrne) [inline continuum-auth/src/use-cases/index.test.ts]: .only here is redundant as there is only one test in the file.

File: continuum-auth/src/utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    describe('check status', () => {
+        it('asserts on ok status', () => {
+            const status = checkStatus(new Response('', { status: 200 }));
+            expect(status).toBeInstanceOf(Response);
+        });
+        it('throws exception when ok status not found', () => {
+            expect(() => {
+                checkStatus(new Response('', { status: 400 }));
+            }).toThrow(new Error('Bad Request'));
+            // todo - check log written to if exception thrown.
```
Comment (will-byrne) [inline continuum-auth/src/utils.test.ts]: we dont want to be checking the logs inside a unit test

Comment (nlisgo) [reply continuum-auth/src/utils.test.ts]: All checking of logs have been removed from the PR.

File: continuum-auth/src/use-cases/index.test.ts (hunk 4 above)
Comment (nlisgo) [reply continuum-auth/src/use-cases/index.test.ts]: Removed. Thanks.

File: continuum-auth/src/jwt.test.ts (hunk 3 above)
Comment (nlisgo) [reply continuum-auth/src/jwt.test.ts]: No longer needed now that log checks have been removed from these tests.

**Comment by nlisgo** [comment]
@Cooryd are you happy with the level of test coverage? 

https://github.com/libero/reviewer/pull/444#issuecomment-549290875

**Pull request merged**